api/src/main/java/ai/djl/training/Trainer.java

```diff
@@ -238,6 +242,15 @@ public Model getModel() {
         return model;
     }
 
+    /**
+     * Returns the {@link ExecutorService}.
+     *
+     * @return the {@link ExecutorService}
+     */
+    public Optional<ExecutorService> getExecutorService() {
+        return Optional.ofNullable(executorService);
```

Comment:
Should we use Optional for all optional arguments?

Reply:
We should probably discuss that as a larger improvement, but I would be in support of it